### ISSUE TITLE
why using the VG data to train the Faster-RCNN model

### ISSUE THREAD

**Issue by yuzcccc** [peteanderson80/bottom-up-attention#2]
Thanks for sharing the models and features. I have tried the feature for VQA with my own model,  really surprising results indeed :)   
 I have two questions as follows: 
1. As the VQA datasets is based on the images of MSCOCO, will it be better to train the faster rcnn model on the COCO dataset directly? 
2. Could a better object detection model, e.g., R-fcn or Deformable R-fcn  further improve the VQA performance?

**Comment by jwyang** [comment]
Hi, @yuzcccc , how this feature help you VQA model? Is it possible to give some numbers from your experiments? I am really interested in this model. Thanks.

**Comment by peteanderson80** [comment]
@yuzcccc 
1. I assume you mean train directly on the 80 COCO object classes? I believe this wouldn't work as well, as the number of object classes is too small. Many interesting objects and regions would be ignored compared to the 1600 objects / 400 attributes that we use from Visual Genome.
2. This could be possible. I think there is a lot more potential to leverge object detection in other tasks, rather than just using plain CNN's as input.

**Comment by jwyang** [comment]
@peteanderson80 , thanks for providing the code. The work is very solid. I am planing to use your code to train a model. Before that, I have several questions as well:

1. How many epochs did you train this model? From your solver file and the statement in your paper, you trained the model for about 2 epochs, right? So it will cost me about 5 days to train the model for 2 epochs if I use other comparable GPUs, like TITAN X, right? I am also wondering what will happen to the performance if you keep training the model. 

2. In the readme, you reported the object detection and attribute classification performance in two cases, one is normal case and one is weighted case. It seems that weighted version makes a lot of difference. Could you explain a bit about the difference between these two cases? I did not find any information about these two setting in your paper. 


thanks in advance!

**Comment by peteanderson80** [comment]
Sure, I will try to help with your questions.
1. When I trained the [final model](https://storage.googleapis.com/bottom-up-attention/resnet101_faster_rcnn_final.caffemodel), I spent about 5 days using 8 gpus (similar to Titan x). The total iterations was 380k. In terms of epochs, each gpu processed 1 image, so I think the model saw about 3M images, or about 29 epochs. However, this is probably overkill, the performance at 320k is about the same, which would be 24 epochs. 
2. The only difference here is the evaluation, the model is the same. Because the object classes are very imbalanced, the weighted case weights each class by it's frequency, instead of just averaging over classes. 
thanks,



./experiments/scripts/faster_rcnn_end2end_multi_gpu_resnet_final.sh

**Comment by jwyang** [comment]
Hi, @peteanderson80 , thanks a lot for your response. I am more clear now.

**Comment by yuzcccc** [comment]
@jwyang @peteanderson80 Using the feature (the model with dynamic K range from [10,100]) extracted from this repo, my MFH model (https://github.com/yuzcccc/mfb/tree/master/mfh_coatt_glove) without the VG data augmentation achieved an accuracy of 68.6% on the VQA-2.0 dataset with a single model :)

**Comment by peteanderson80** [comment]
Thanks @yuzcccc. For interest, what was the accuracy for the same model using standard CNN features (e.g. ResNet)?

**Comment by yuzcccc** [comment]
@peteanderson80 
66.12% for the 14*14 res5c layer extracted from ResNet-152

**Comment by peteanderson80** [comment]
Cool, thanks for the info. I think our relative improvement was a little larger, but we also optimized hyperparameters for the new features.
